fenix/app/src/main/java/org/mozilla/fenix/components/StoreProvider.kt

```diff
@@ -37,23 +48,36 @@ class StoreProvider<T : Store<*, *>>(
  */
 @VisibleForTesting
 class StoreProviderFactory<T : Store<*, *>>(
-    private val createStore: () -> T,
+    private val createStore: (CoroutineScope) -> T,
 ) : ViewModelProvider.Factory {
 
     @Suppress("UNCHECKED_CAST")
     override fun <VM : ViewModel> create(modelClass: Class<VM>): VM {
-        return StoreProvider(createStore()) as VM
+        return StoreProvider(createStore) as VM
     }
 }
 
 /**
  * Helper function for lazy creation of a [Store] instance scoped to a [ViewModelStoreOwner].
  *
- * @param createStore Function that creates a [Store] instance.
+ * @param createStore Function that creates a [Store] instance that receives [CoroutineScope] param
+ * that's tied to the lifecycle of the [StoreProvider] i.e it will cancel when
+ * [StoreProvider.onCleared] is called.
+ *
+ * @sample
+ * val store by lazy { scope ->
+ *   MyStore(
+ *     middleware = provideMyMiddleware(
+ *     settings = requireComponents.settings,
+ *     ...
+ *     scope = scope,
+ *     ),
+ *   )
+ * }
```

Comment:
It looks like `@sample` requires an identifier to a function. Instead, let's just use a regular code block for now.

Separate nit: use a `listOf(middleware)` in the sample.

```suggestion
 * Example:
 * ```kotlin
 * val store by lazy { scope ->
 *   MyStore(
 *     middleware = listOf(
 *       MyMiddleware(
 *         settings = requireComponents.settings,
 *         ...
 *         scope = scope,
 *       ),
 *     )
 *   )
 * }
 * ```
```